### ISSUE TITLE
4.4dev3 Regression: `MeshInstance3D.skeleton` bindings are lost when removing/adding nodes to scene tree

### ISSUE THREAD

**Issue by HauntedBees** [godotengine/godot#98349]
### Tested versions

- Reproducible in: 4.4.dev3.official [f4af8201b].
- Not reproducible in: 4.3.stable.official [77dcf97d8].

### System information

Linux Mint 21.1 - Godot v4.4.dev3 - Compatibility

### Issue description

I have a scene with this structure -- namely a `Skeleton3D` with several `MeshInstance3D` nodes under it, which all have a `skeleton` property set to the parent `Skeleton3D` in the Godot Editor.
![image](https://github.com/user-attachments/assets/e8c1fdc4-302c-4cdf-aa8b-74af357b207b)

If I use `remove_child` to remove the `Skeleton3D` or any of its parent nodes from the scene tree, and then re-add that node with `add_child`, the `skeleton` property's value is lost, and needs to be manually corrected with a method like this attached to the script of the `Skeleton3D`:

```gdscript
func _enter_tree() -> void:
	var skel_path := get_path()
	for child in get_children():
		var mesh := child as MeshInstance3D
		if mesh:
			mesh.skeleton = skel_path
```

### Steps to reproduce

Use `remove_child` to remove a `Skeleton3D` node or a node that has a `Skeleton3D` node as a descendant from the Scene Tree, then use `add_child `to add the removed node back. Try to play an animation bound to the skeleton; the animation will play but the skeleton/meshes will not animate.

### Minimal reproduction project (MRP)

1. Run the test project/scene. Click the "Start Animation" button to confirm the animation works.
2. Click the "Remove Character" and "Add Character" buttons, then click "Start Animation" again to confirm that animations are no longer working, as the skeleton bindings are lost.
3. Click the "Manually Fix Bindings" button to correct this, then watch that "Start Animation" now works as expected.

[skeleton-test.zip](https://github.com/user-attachments/files/17447618/skeleton-test.zip)

**Comment by matheusmdx** [comment]
This regression was caused by pr #97489 and was already fixed by pr #98009

**Comment by HauntedBees** [comment]
That works for me! I'm looking forward to 4.4dev4 then! Thank you!
